confirm.c

```diff
@@ -763,9 +855,35 @@ confirm_query_create4(const struct sockaddr_storage *dst, uint8_t ttl,
 }
 
 struct confirm_query *
+confirm_query_create6_tcp(const struct sockaddr_storage *dst, uint8_t ttl,
+		uint8_t traffic_class, uint32_t flow_label, uint8_t flowid,
+		uint16_t src_port, uint16_t dst_port, confirm_query_cb cb)
+{
+	struct confirm_query *query = confirm_query_create6(dst, ttl, traffic_class,
+			flow_label, flowid, cb);
+
+	query->src_port = src_port;
+	query->dst_port = dst_port;
+	query->type = QUERY_TYPE_TCP;
+	return query;
+}
+
+struct confirm_query *
+confirm_query_create6_icmp(const struct sockaddr_storage *dst, uint8_t ttl,
+		uint8_t traffic_class, uint32_t flow_label, uint16_t icmpid,
+		uint8_t flowid, confirm_query_cb cb)
+{
+	struct confirm_query *query = confirm_query_create6(dst, ttl, traffic_class,
+			flow_label, flowid, cb);
+
+	query->icmpid = icmpid;
+	query->type = QUERY_TYPE_ICMP;
+	return query;
+}
+
+struct confirm_query *
 confirm_query_create6(const struct sockaddr_storage *dst, uint8_t ttl,
-		uint8_t traffic_class, uint32_t flow_label,
-		uint16_t icmpid, uint8_t flowid,
+		uint8_t traffic_class, uint32_t flow_label, uint8_t flowid,
```

Comment:
if it is not actively used, I would try to remove this function and call the `_icmp` and `_tcp` versions. (i don't remember what else is in the code though)